### ISSUE TITLE
Training issue

### ISSUE THREAD

**Issue by maollila** [zhaoyuzhi/RAW2RGB-GAN#3]
Hello,

I have succesfully used test code with trained weights and datasets provided by you.
python test.py --netroot "zyz987.pth" --baseroot "C:\imagesets\Zurich-RAW-to-DSLR-Dataset\TestingPhoneRaw" --saveroot "TestingResults"
I am using below training commands.
First stage 1:
python train.py --in_root "C:\imagesets\Zurich-RAW-to-DSLR-Dataset\TestingPhoneRaw" --out_root "C:\imagesets\Zurich-RAW-to-DSLR-Dataset\TrainingCanonRGB" --sal_root "C:\imagesets\Zurich-RAW-to-DSLR-Dataset\TrainingCanonRGB_saliency\TrainingCanonRGB_saliency"
second stage 2:
python train.py --pre_train False --epochs 30 --lr_g 0.0001 --in_root "C:\imagesets\Zurich-RAW-to-DSLR-Dataset\TestingPhoneRaw" --out_root "C:\imagesets\Zurich-RAW-to-DSLR-Dataset\TrainingCanonRGB" --sal_root "C:\imagesets\Zurich-RAW-to-DSLR-Dataset\TrainingCanonRGB_saliency\TrainingCanonRGB_saliency" 

I added --pre_train False not mentioned instructions. 

I  used same test command but used resulting Pre_RAW2RGB_epoch30_bs8.pth.

environment is pytorch 1.11 and python 3.9.
![0](https://user-images.githubusercontent.com/108550429/185347636-2955d40a-1166-4e6b-bf9b-a60fb0ac06be.png)

Resulted image is not ok. Not a color image. see attached image.

**Comment by zhaoyuzhi** [comment]
Hi,

I test the code on a randomly selected image (6.png) in TestingPhoneRaw, the pre-trained model can be found at this link: https://github.com/zhaoyuzhi/RAW2RGB-GAN#12--test

The input is:
![6](https://user-images.githubusercontent.com/13333802/185358653-e292ff0d-cbce-47e3-8829-8f9092ee1653.png)
The resulting image is like this:
![6](https://user-images.githubusercontent.com/13333802/185358544-0131bf78-a0a6-4c25-b45b-7efc25fc5255.png)

It seems that the pre-trained weights are correct. I am not sure whether your environment is good for this project.

I test the code on CUDA 9.0, PyTorch 1.0, and python 3.6.

**Comment by maollila** [comment]
Thank you for very quick answer. Your pretrained weights work well also in my environment.  What I mean here is that training does not seem to work.

**Comment by zhaoyuzhi** [comment]
Sorry, I also do not know where is the problem based on this information. Actually, the training codes are totally the same as those I submit to the ICCV workshop organizers. I recommend you directly use the provided model if you want to do some experiments and use it for comparisons.

By the way, could you show some prints/logs (e.g., `Pixellevel L1 Loss`) during the training, this would be helpful for identifying where is the problem? It might be helpful for debugging.

**Comment by maollila** [comment]
[pytorch1v0cuda90py36stage1.txt](https://github.com/zhaoyuzhi/RAW2RGB-GAN/files/9374523/pytorch1v0cuda90py36stage1.txt)
this stage  1 shows logs of both pytorch1v00cuda90python3.6 that fails and then cuda 11.3 pytorch 1.11 py39. Pytorch1v00 fails and then you can see logs of stage 1 for pytorch1.11.

**Comment by maollila** [comment]
[stage2.txt](https://github.com/zhaoyuzhi/RAW2RGB-GAN/files/9376904/stage2.txt)
This stage 2 logs

**Comment by maollila** [comment]
When I try you pretrained weights that do produce good color images even with Pytorch 1.11 + Cuda 11.3 + Py39, I get these warnings.  Does this hint something needs to be changed in Networks for Pytorch 1.11?

(pytorch) C:\Users\MikkoOllila\Documents\docs\Computer_vision\ISP\demosaicking\deep\RAW2RGB-GAN-master\RAW2RGB-GAN-master>python test.py --netroot "zyz987.pth" --baseroot "C:\imagesets\Zurich-RAW-to-DSLR-Dataset\TestingPhoneRaw" --saveroot "TestingResults"
C:\Users\MikkoOllila\anaconda3\envs\pytorch\lib\site-packages\torch\serialization.py:786: SourceChangeWarning: source code of class 'torch.nn.modules.padding.ReflectionPad2d' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
C:\Users\MikkoOllila\anaconda3\envs\pytorch\lib\site-packages\torch\serialization.py:786: SourceChangeWarning: source code of class 'torch.nn.modules.activation.LeakyReLU' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
C:\Users\MikkoOllila\anaconda3\envs\pytorch\lib\site-packages\torch\serialization.py:786: SourceChangeWarning: source code of class 'torch.nn.modules.conv.Conv2d' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
C:\Users\MikkoOllila\anaconda3\envs\pytorch\lib\site-packages\torch\serialization.py:786: SourceChangeWarning: source code of class 'torch.nn.modules.instancenorm.InstanceNorm2d' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
C:\Users\MikkoOllila\anaconda3\envs\pytorch\lib\site-packages\torch\serialization.py:786: SourceChangeWarning: source code of class 'torch.nn.modules.container.Sequential' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
C:\Users\MikkoOllila\anaconda3\envs\pytorch\lib\site-packages\torch\serialization.py:786: SourceChangeWarning: source code of class 'torch.nn.modules.activation.Tanh' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
C:\Users\MikkoOllila\anaconda3\envs\pytorch\lib\site-packages\torch\serialization.py:786: SourceChangeWarning: source code of class 'torch.nn.modules.activation.Sigmoid' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)

**Comment by zhaoyuzhi** [comment]
Hi,

I think the training at the very beginning failed.

The Pixellevel L1 Loss should be around 0.1......(or maybe 0.2....... or 0.0..........), but in your txt file, it starts with 0.6161. You might check whether the input/output is paired by visualizing them. Or you might change the initialization method/normalization of images.

As for the warning, I assume that it would not be a problem. Maybe you can refer to this: https://github.com/tbepler/protein-sequence-embedding-iclr2019/issues/17

**Comment by maollila** [comment]
Hi,

Thank you for your good suggestion. I think I have found an error as images are indeed not paired as I am erroneously using directory of ZRR raw not based on your paired directories.  I think I have tomorrow good news after training has completed.

**Comment by maollila** [comment]
Hi,

It works. Thank you very much. Did I understand correctly that stage 2 training uses some epoch from stage 1?  --pre_train', type = bool, default = True, help = 'pre-train ot not') # for second stage, change it to False --> this needs to be set False for stage 2. This was not in instructions, but in py file.
